15.6 error handling

kosara-g opened the pull request:
This is what I have so far for the error handling of the server. I'm unable to verify it works with develop currently as there are issues in said branch. Suggestions on how to proceed are very, very welcome.

Commit:
Added some linking between the processes

Commit:
Added collaborator tag

Commit:
Deleted unused code

Commit:
Fixed if loop that cause trouble

pontuslaestadius commented on Server/src/diagramcoordinator.erl:
```diff
@@ -2,7 +2,8 @@
 %%Author: Tim Jonasson
-%%Version: 1.3
+%%Version: 1.4
+%%@collaborator Kosara Golemshinska
```
Don't use a @ it's not javadocs

pontuslaestadius commented on Server/src/node.erl:
```diff
@@ -2,11 +2,13 @@
 %%Author: Tim Jonasson
-%%Version: 1.2
+%%Version: 1.3
+%%@collaborator Kosara Golemshinska
```
Same as previous. remove @

pontuslaestadius commented on Server/src/usercoordinator.erl:
```diff
@@ -2,15 +2,26 @@
 %%Author: Tim Jonasson
-%%Version: 2.1
+%%Version: 2.2
+%%@collaborator Kosara Golemshinska
```
remove @ please!

pontuslaestadius reviewed (commented):
I'm not qualified to judge if this is fine or not.

Commit:
Removed @ as requested

timjon commented on Server/src/diagramcoordinator.erl:
```diff
@@ -45,7 +46,11 @@ spawn_nodes(List) ->[spawn_node(Class) || Class <- List].
+  % Link the node process to the diagram coordinator so
+  % that if the node crashes, the coordinator also crashes.
+  link(Pid),
+  {Pid, Class_name}.
```
these changes shouldnt be necessary since you are using spawn_link in node

timjon commented on Server/src/usercoordinator.erl:
```diff
@@ -36,7 +47,13 @@ loop(Socket, Diagrams) ->
+	% If the user coordinator process gets an exit
+ 	% message, it either terminates, or restarts
+	% itself based on the exit reason.
+	{'EXIT', _From, normal} -> ok;
```
Is the indentation weird on git or is it an error?

timjon commented on Server/src/usercoordinator.erl:
```diff
@@ -2,15 +2,26 @@
+  % If the diagram list has at least 1 item,
+  % get its Pid and link to it. This way, error
+  % propagation is ensured. 
+  if length(Diagrams) > 0 ->
```
Is this needed?

kosara-g replied:
Yes, if the list is empty, hd/1 will throw an error

timjon commented on Server/src/usercoordinator.erl:
```diff
@@ -36,7 +47,13 @@ loop(Socket, Diagrams) ->
+ 	% message, it either terminates, or restarts
+	% itself based on the exit reason.
+	{'EXIT', _From, normal} -> ok;
+    {'EXIT', _P, _Reason} -> loop(Socket, Diagrams) 
```
You need to remove the crashed diagramcoordinator from the Diagrams list and then readd it

Commit:
Version change, trying to resolve conflicts with develop

Commit:
Made changes according to request

IsabelleTornqvist commented on Server/src/diagramcoordinator.erl:
```diff
@@ -2,17 +2,20 @@
-  Pids = spawn_nodes(L),
-  loop(Usercoordinator, Did, Pids, Messages, 1).
-  
+	%Sending information that the usercoordinator has been spawned. To be printed in the executionlog
```
These are changes done for the executionlog. what will happen when merging this?

kosara-g replied:
I didn't change this, this was taken from develop

IsabelleTornqvist replied:
ah oki

Commit:
Changes as per request

Removed timeout and added the classes and messages to the Diagrams list
to be able to restart the user coordinator.

kosara-g commented:
Insufficient time to finish the integration with the current release.
